source/agora/flash/OnionPacket.d

```diff
@@ -512,3 +512,49 @@ public struct OnionError
     /// Error code
     ErrorCode err;
 }
+
+/*******************************************************************************
+
+    Obfuscate/deobfuscate an OnionError with the given secret.
+
+    Params:
+        error = Error to obfuscate/deobfuscate
+        secret = Shared secret
+
+    Returns:
+        Obfuscated/deobfuscated OnionError
+
+*******************************************************************************/
+
+public OnionError obfuscate (OnionError error, Point secret) @trusted
+{
+    import libsodium.crypto_stream_chacha20;
+
+    const key = hashFull(secret);
+    assert(key[].length >= crypto_stream_chacha20_KEYBYTES);
+    assert(error.payment_hash[].length >= crypto_stream_chacha20_NONCEBYTES);
+
+    ubyte[error.sizeof] stream;
```

Comment:
OnionError.sizeof specifically. Just to make it clear it's fixed-size.

Reply:
Done